app/helpers/templates_helper.rb

```diff
@@ -46,4 +46,19 @@ def locked_warning(template)
 
     alert(:class => 'alert-warning', :text => warning_text.html_safe)
   end
+
+  def self.template_help_text
+    {
+      "PXELinux" => N_("Deployed to the TFTP server to ensure the Host boots the correct installer using pxelinux.0 file"),
+      "PXEGrub" => N_("Deployed to the TFTP server to ensure the Host boots the correct installer using grub/bootx64.efi file"),
+      "PXEGrub2" => N_("Deployed to the TFTP server to ensure the Host boots the correct installer using grub2/grubx64.efi file"),
+      "iPXE" => N_("Used in {g,i}PXE environments in place of PXELinux"),
+      "provision" => N_("The main unattended installation file; e.g. Kickstart or Preseed"),
+      "finish" => N_("A post-install script used to take custom actions after the main provisioning is complete"),
+      "script" => N_("An arbitrary script, not used by default, useful for certain custom tasks"),
+      "user_data" => N_("Similar to a Finish script, this can be assigned to hosts built on user_data-capable images (e.g. Openstack, EC2, etc)"),
```

Comment:
Openstack, EC2 are not images but providers, perhaps mention cloud-init as the tool that fetches user_data?

Reply:
Also the checkbox must be checked in order to use user-data. This can get pretty long, maybe we can detect it?

Reply:
I wouldn't worry about the length, it's hidden in tooltip now. I think it's better to just describe and don't change help dynamically.

Reply:
Perhaps:

Template with seed data for virtual or cloud instances with "user data" flag set, typically cloud-init or ignition format.